app/controllers/user.js

```diff
@@ -77,19 +77,26 @@ exports.buyAlbum = (req, res, next) =>
 exports.listAlbumsUser = (req, res, next) => {
   const keyValue = { userId: req.params.user_id };
 
-  const getAlbums = userAlbums => {
-    const albumsData = [];
-    for (const userAlbum of userAlbums) {
-      albumsData.push(albumService.getAlbumById(userAlbum.albumId));
-    }
-    return Promise.all(albumsData);
-  };
   return userService
     .findBoughtAlbums(keyValue)
     .then(boughtAlbums => {
-      getAlbums(boughtAlbums).then(albumsData => {
+      const albums = boughtAlbums.map(album => albumService.getAlbumById(album.albumId));
+      return Promise.all(albums).then(albumsData => {
         res.send({ albumsData });
       });
     })
     .catch(next);
 };
+
+exports.listPhotosAlbumsBought = (req, res, next) =>
+  userService
+    .findBoughtAlbums({ albumId: req.params.id })
+    .then(albums => {
```

Comment:
This should be only one combination of userId and albumId

Reply:
Like this any user can list photos of any other user, i don't see any filter for that